lib/ingress/permission_rule.rb

```diff
@@ -14,15 +18,26 @@ def allows?
     end
 
     def match?(given_action, given_subject, user, options = {})
+      validate_given_subject!(given_subject, conditions)
+
       return false unless action_matches?(given_action)
       return false unless subject_matches?(given_subject)
-      return true if ignore_conditions?(given_subject)
 
       conditions_match?(user, given_subject, options)
     end
 
     private
 
+    def validate_given_subject!(subject, conditions)
+      has_some_conditions =  !(conditions.nil? || conditions.size == 0)
```

Comment:
Purely a style suggestion (feel free to ignore):

```suggestion
      return if conditions.nil? || conditions.size == 0
      return unless subject_class?(subject)
      
      raise_error!
```